### PR TITLE
MMDP- 323 Media grid list (video/images)

### PR DESCRIPTION
**Pull request by p8ul**
**What does this PR do?**

- Implement repository media grid list

**Description of Task to be completed?**

- Implement repository media grid list (Video/images)

**How should this be manually tested?**

- Install dependencies `yarn install`
- Run server `yarn dev-server`
- Then navigate to the following path

|Description    | Path   | 
| ------------- | ------------- 
| List Media |{{url}}/resources/media}
 

**Any background context you want to provide?**

- None

**What are the relevant Jira issues?**

[MMDP-323](https://viisaus.atlassian.net/browse/MMDP-323)

**Screenshots (if appropriate)**
screenshot 2019-02-20 at 10 49 35

**Questions:**

- None

### CODE REVIEW

File: client/components/Resources/Document/DocumentComponent.js [hunk 1]
```diff
@@ -1,14 +1,25 @@
 import React from 'react';
```
Comment (shemogumbe) [inline client/components/Resources/Document/DocumentComponent.js]: If we are building a component, and it's already inside components/ folder, do not call it DocumentComponent, just call it Document as the context is already clear

File: client/components/Resources/Document/DocumentComponent.js [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+import PropTypes from 'prop-types';
 import SearchMedia from './SearchMedia';
-import MediaList from './MediaList';
+import DocumentList from './MediaList';
+import MediaType from '../Media/MediaType';
 
 const DocumentComponent = (props) => {
```
Comment (shemogumbe) [inline client/components/Resources/Document/DocumentComponent.js]: Rename component to Document

File: client/components/Resources/Document/DocumentForm.js [hunk 3]
```diff
@@ -28,10 +28,10 @@ const DocumentForm = ({
         name="title"
         className="form__text-input"
         fluid
-        label="Report title"
+        label="Document title"
```
Comment (shemogumbe) [inline client/components/Resources/Document/DocumentForm.js]: Rename to AddDocument, and not DocumentForm, the context Add will communicate what the component does better that the word Form

File: client/components/Resources/Document/MediaList.js [hunk 4]
```diff
@@ -1,15 +1,21 @@
...
+import DocumentCard from './MediaCard';
+import MediaCard from '../Media/MediaCard';
+
 import InvalidPage from '../../common/InvalidPage';
 
-const MediaList = ({ loading, documents, goTo }) => (
+const MediaList = ({ loading, documents, goTo, isMedia }) => (
   <Grid columns={4} className="ui loading" loading={loading.toString()}>
     <Grid.Row className="animated fadeIn">
-      {documents.results.map((item) => (
```
Comment (shemogumbe) [inline client/components/Resources/Document/MediaList.js]: The media component ought tp have pagination, use the common pagination component already created to and render media cards within it

File: client/components/Resources/Document/SearchMedia.js [hunk 5]
```diff
@@ -14,15 +15,23 @@ const SearchMedia = () => {
           </Button>
         </Grid.Column>
         <Grid.Column width={3}>
```
Comment (shemogumbe) [inline client/components/Resources/Document/SearchMedia.js]: A search component already exists as a common, use it and write a search handler for  the search button

File: client/components/Resources/Document/DocumentForm.js (hunk 3 above)
Comment (p8ul) [reply client/components/Resources/Document/DocumentForm.js]: The component caters for both adding and editing context.

File: client/components/Resources/Document/MediaList.js (hunk 4 above)
Comment (p8ul) [reply client/components/Resources/Document/MediaList.js]: Search and pagination for media listing is a different story. This story was about media grid listing.

**Pull request merged**